### PR TITLE
Avoid floating point comparison issues

### PR DESCRIPTION
**Pull request by gonzaponte**
Some tests that check for an "exact result" have been problematic because they rely on floating point comparisons. This is machine-dependent, which means a test may pass locally, but fail on GHA or viceversa. With direct comparisons, this is usually addressed by adding a tolerance, which we already do. However there are other cases where this isn't enough
- cumulative rounding errors: typical of beersheba, where the iterative process seems to accumulate errors beyond what one might expect using doubles (?)
- discretization of continuous values: typical of paolina, where voxelization is sensible to tiny differences between floating point values, despite our attempts to avoid them.

In this PR, we bypass all these issues by rounding values to a safe number while maintaining enough precision.

### CODE REVIEW

**Review by Luismiguelvillar — COMMENTED** [review]
It looks good to me. I just have a question regarding the chosen precision. Is there a specific reason for rounding to 6 decimal places? 

`deco_dst.E = np.round(deco_dst.E, 6)`

**Comment by gonzaponte** [comment]
Not a fundamental reason, but a practical one. Rounding to more digits produced discrepancies between my computer and GHA. 6 digits seems enough to maintain precision while avoiding these issues.

File: invisible_cities/reco/paolina_functions_test.py [hunk 1]
```diff
@@ -165,6 +166,24 @@ def sum_energy(seq):
     assert sum_energy(voxels) == approx(sum_energy(hits))
 
 
+@given(bunch_of_hits)
+def test_round_hits_positions_in_place(hits):
```
Comment (Luismiguelvillar) [inline invisible_cities/reco/paolina_functions_test.py:167]: Should we also test how the function behaves with edge cases such as NaNs, infinities, or empty hit lists?

**Comment by Luismiguelvillar** [comment]
> Not a fundamental reason, but a practical one. Rounding to more digits produced discrepancies between my computer and GHA. 6 digits seems enough to maintain precision while avoiding these issues.

Makes sense! But maybe it will be a good idea to let the user change this precision, similar to the xyz in  hits?

File: invisible_cities/reco/paolina_functions.py [hunk 2]
```diff
@@ -32,6 +32,14 @@ def bounding_box(seq : BHit) -> Sequence[np.ndarray]:
     return (reduce(np.minimum, posns, MAX3D),
             reduce(np.maximum, posns, MIN3D))
 
+def round_hits_positions_in_place(hits, decimals=5):
```
Comment (Luismiguelvillar) [inline invisible_cities/reco/paolina_functions.py:35]: It looks good, and modifying the hits in place was a good choice.
My only minor concern is that the function is not protected against NaNs, infinities, or other non-finite values. This is not particularly critical, since `np.round()` will simply leave those non-finite values unchanged (furthermore, I do not think hits with non-finite coordinates will make it to this point).

Comment (gonzaponte) [reply invisible_cities/reco/paolina_functions.py:35]: I guess the expected result in that case (that shouldn't happen, but I've seen worse) is to leave those values unchanged. I'll add dedicated tests as requested above.

**Review by Luismiguelvillar — APPROVED** [review]
Everything looks fine!

**Comment by gonzaponte** [comment]
@mcidlaso please merge
